src/create-matcher.js

```diff
@@ -65,9 +65,9 @@ export function createMatcher (routes: Array<RouteConfig>): Matcher {
         hash,
         params
       }, undefined, location)
-    } else if (typeof redirect === 'string') {
+    } else if (typeof redirect === 'string' || typeof redirect === 'function') {
       // 1. resolve relative redirect
-      const rawPath = resolveRecordPath(redirect, record)
+      const rawPath = resolveRecordPath(typeof redirect === 'function' ? redirect() : redirect, record)
```

Comment:
It doesn't have an impact on performance so I think it's ok this way.

However, I think it's better to call redirect if it's a function and let the rest of the code the same:

`redirect = typeof redirect === 'function' ? redirect() : redirect`

This way the return of redirect can also be an object with a name and it's more flexible